From 26fd892066f510c5e329b27f3217bd3473423f43 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Sat, 3 Mar 2018 14:54:59 +0700
Subject: [PATCH] Problem: commenting on an issue with an empty text doesn't
 work

Scenario:

1. Open an issue
2. Scroll to the bottom
3. Press "Comment" (or "Comment and close")

Outcome: silence and an error in console.log (HTTP 500 for "Comment" and
"Uncaught TypeError: Cannot read property 'append' of undefined" for
"Comment and close")

Solution: disable commenting buttons unless there's text to comment with
---
 sit-web/webapp/src/issue/new-comment.html | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/sit-web/webapp/src/issue/new-comment.html b/sit-web/webapp/src/issue/new-comment.html
index 74f439d..5401868 100644
--- a/sit-web/webapp/src/issue/new-comment.html
+++ b/sit-web/webapp/src/issue/new-comment.html
@@ -12,12 +12,12 @@
 
         <div>
             <div>
-              <paper-textarea label="Comment" on-keypress="handleChange" on-change="handleChange" on-input="handleChange" id="comment" cols="80" rows="5"></paper-textarea>
+              <paper-textarea label="Comment" on-keypress="handleChange" on-change="handleChange" on-input="handleChange" id="comment" cols="80" rows="5" auto-validate required></paper-textarea>
             </div>
             <div>
-                <paper-button raised id="commentButton" on-click="handleComment">Comment</paper-button>
+                <paper-button raised id="commentButton" on-click="handleComment" disabled="[[ invalid ]]">Comment</paper-button>
                 <template is="dom-if" if="[[ isOpen ]]">
-                    <paper-button raised on-click="handleCommentAndClose">Comment and close</paper-button>
+                    <paper-button raised on-click="handleCommentAndClose" disabled="[[ invalid ]]">Comment and close</paper-button>
                 </template>
             </div>
         </div>
@@ -50,6 +50,10 @@
                     },
                     config: {
                         type: Object
+                    },
+                    invalid: {
+                        type: Boolean,
+                        value: true,
                     }
                 }
             }
@@ -61,6 +65,7 @@
                 return author;
             }
             handleChange() {
+                this.set('invalid', this.$.comment.invalid);
                 let timestamp = (new Date()).toISOString();
                 let formdata = new FormData();
                 formdata.append("text", new Blob([this.$.comment.value], {type: "text/plain"}), "text");
-- 
2.15.0

